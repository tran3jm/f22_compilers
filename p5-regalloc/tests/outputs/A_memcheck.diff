--- outputs/A_memcheck.txt	2022-12-04 17:23:13.022710261 -0500
+++ expected/A_memcheck.txt	2022-11-20 13:50:39.000000000 -0500
@@ -1,14 +1,3 @@
-Handling main:
-Handling push BP
-Handling i2i SP => BP
-Handling addI SP, 0 => SP
-Handling loadI 4 => r0
-Handling i2i r0 => RET
-Handling jump l0
-Handling l0:
-Handling i2i BP => SP
-Handling pop BP
-Handling return
 main:
   push BP
   i2i SP => BP
@@ -20,4 +9,85 @@
   i2i BP => SP
   pop BP
   return
+
+==========================
+sp=65536 bp=-9999999 ret=-9999999
+virtual regs: 
+stack:
+other memory:
+==========================
+
+Executing: push BP
+
+==========================
+sp=65528 bp=-9999999 ret=-9999999
+virtual regs: 
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: i2i SP => BP
+
+==========================
+sp=65528 bp=65528 ret=-9999999
+virtual regs: 
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: addI SP, 0 => SP
+
+==========================
+sp=65528 bp=65528 ret=-9999999
+virtual regs: 
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: loadI 4 => r0
+
+==========================
+sp=65528 bp=65528 ret=-9999999
+virtual regs:  r0=4
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: i2i r0 => RET
+
+==========================
+sp=65528 bp=65528 ret=4
+virtual regs:  r0=4
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: jump l0
+
+==========================
+sp=65528 bp=65528 ret=4
+virtual regs:  r0=4
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: i2i BP => SP
+
+==========================
+sp=65528 bp=65528 ret=4
+virtual regs:  r0=4
+stack:  65528: -9999999
+other memory:
+==========================
+
+Executing: pop BP
+
+==========================
+sp=65536 bp=-9999999 ret=4
+virtual regs:  r0=4
+stack:
+other memory:  65528: -9999999
+==========================
+
+Executing: return
 RETURN VALUE = 4
