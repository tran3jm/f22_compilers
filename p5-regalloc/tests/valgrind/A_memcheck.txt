==1181128== Memcheck, a memory error detector
==1181128== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1181128== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==1181128== Command: ../decaf inputs/sanity.decaf
==1181128== 
Handling main:
Handling push BP
Handling i2i SP => BP
Handling addI SP, 0 => SP
Handling loadI 4 => r0
Handling i2i r0 => RET
Handling jump l0
Handling l0:
Handling i2i BP => SP
Handling pop BP
Handling return
main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 4 => r0
  i2i r0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return
RETURN VALUE = 4
==1181128== 
==1181128== HEAP SUMMARY:
==1181128==     in use at exit: 0 bytes in 0 blocks
==1181128==   total heap usage: 1,779 allocs, 1,779 frees, 461,580 bytes allocated
==1181128== 
==1181128== All heap blocks were freed -- no leaks are possible
==1181128== 
==1181128== For lists of detected and suppressed errors, rerun with: -s
==1181128== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
