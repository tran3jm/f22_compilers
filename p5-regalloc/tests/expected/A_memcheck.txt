main:
  push BP
  i2i SP => BP
  addI SP, 0 => SP
  loadI 4 => r0
  i2i r0 => RET
  jump l0
l0:
  i2i BP => SP
  pop BP
  return

==========================
sp=65536 bp=-9999999 ret=-9999999
virtual regs: 
stack:
other memory:
==========================

Executing: push BP

==========================
sp=65528 bp=-9999999 ret=-9999999
virtual regs: 
stack:  65528: -9999999
other memory:
==========================

Executing: i2i SP => BP

==========================
sp=65528 bp=65528 ret=-9999999
virtual regs: 
stack:  65528: -9999999
other memory:
==========================

Executing: addI SP, 0 => SP

==========================
sp=65528 bp=65528 ret=-9999999
virtual regs: 
stack:  65528: -9999999
other memory:
==========================

Executing: loadI 4 => r0

==========================
sp=65528 bp=65528 ret=-9999999
virtual regs:  r0=4
stack:  65528: -9999999
other memory:
==========================

Executing: i2i r0 => RET

==========================
sp=65528 bp=65528 ret=4
virtual regs:  r0=4
stack:  65528: -9999999
other memory:
==========================

Executing: jump l0

==========================
sp=65528 bp=65528 ret=4
virtual regs:  r0=4
stack:  65528: -9999999
other memory:
==========================

Executing: i2i BP => SP

==========================
sp=65528 bp=65528 ret=4
virtual regs:  r0=4
stack:  65528: -9999999
other memory:
==========================

Executing: pop BP

==========================
sp=65536 bp=-9999999 ret=4
virtual regs:  r0=4
stack:
other memory:  65528: -9999999
==========================

Executing: return
RETURN VALUE = 4
